src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -243,13 +260,50 @@ export const scanAndRetrieveAttachments = async (
       error: triggerLambdaResult.error,
     })
 
-    return res.status(StatusCodes.INTERNAL_SERVER_ERROR).send({
-      message:
-        'Something went wrong while scanning your attachments. Please try again.',
+    const { statusCode, errorMessage } = mapRouteError(
+      triggerLambdaResult.error,
+    )
+    return res.status(statusCode).json({
+      message: errorMessage,
     })
   }
 
-  // TODO(FRM-1302): Step 4: Retrieve attachments from the clean bucket.
+  // Step 4: Retrieve attachments from the clean bucket.
+  const downloadCleanFilesResult = await ResultAsync.combine(
+    triggerLambdaResult.value.map((isAttachment) => {
+      if (isAttachment) {
+        // Retrieve attachment from clean bucket.
+        return downloadCleanFile(
+          isAttachment.cleanFile.cleanFileKey,
+          isAttachment.cleanFile.destinationVersionId,
+        ).map((result) => {
+          // Replace content with file retreived from clean bucket.
+          isAttachment.response.content = result
+          // Replace answer with filename.
+          isAttachment.response.answer = isAttachment.response.filename
+          return true // Return true to indicate that the download was successful.
+        }) // If any downloads error out, it will short circuit and return the first error.
+      }
+
+      // If response is not an attachment, return okAsync(false).
+      return okAsync(false as const)
+    }),
+  )
```

Comment:
I think you can combine the two `Array.prototype.map()` calls in Step 3 and 4 into one, and then return a clean `response[]` at the end of `.map()` in order to replace `req.body.responses`. Then you dont need to return `false as const` along the way. Might also save on performance because you don't need all the virus scans to finish before you start downloading file.